### DISCUSSION TITLE
Is it a little bit too much on asset file thumbnail generation ?

### DISCUSSION THREAD

**Discussion by junbach** [directus/directus#6626] [Archive]
Hello,
Currently I am running directus with storage adapter on S3, and everything works well. But I notice something strange when look at the bucket that the app is using to store asset files. It generates many big size of my uploaded files, even the small one, and it consumes a lot of bucket storage. For example:

- Original file [4.7KB] f6de14a1-99a3-4992-add1-4e8b15bdc583.png
- Generation_01 [12.8KB] f6de14a1-99a3-4992-add1-4e8b15bdc583__fit_cover,height_64,key_system-small-cover,width_64.png
- Generation_02 [835.4KB] f6de14a1-99a3-4992-add1-4e8b15bdc583__fit_cover,height_600,key_system-large-cover,width_800.png

With this files generation, it would take around x180 times more spaces in my storage to store just a small file. And since there may be many files ( one for each record), the spaces needs for just file generation would be very big. Does it work as expected? Are there any workaround or solutions for that?

Many thanks

**Comment by rijkvanzanten** [comment]
Odd! What's the original file? Does it have to scale _up_ to get to the larger files?

**Reply by junbach** [reply]
Original file is a small png file with size 80x80 pixel. This is a sample (4.7KB) : https://cms-assets.wikie.app/f6de14a1-99a3-4992-add1-4e8b15bdc583.png
Yeah, I think the app scale up the image file for some reasons.

And the biggest generated one (835KB) : https://cms-assets.wikie.app/f6de14a1-99a3-4992-add1-4e8b15bdc583__fit_cover,height_600,key_system-large-cover,width_800.png

**Reply by rijkvanzanten** [reply]
> Yeah, I think the app scale up the image file for some reasons.

Ah yeah that makes total sense. It's scaling the 80x80 image up to 800x600 for certain thumbnails 👍🏻 

Could be a nice improvement to skip the size transformation if the image itself is smaller than the requested size 🤔

**Reply by junbach** [reply]
> Could be a nice improvement to skip the size transformation if the image itself is smaller than the requested size 🤔'

Yeah I think this is a good improvement . It can save a lot of disk space, and cost also in case of using object storage
